samza-core/src/main/java/org/apache/samza/zk/ZkBarrierForVersionUpgrade.java

```diff
@@ -105,30 +97,24 @@ public void handleChildChange(String parentPath, List<String> currentChildren) t
         LOG.info("Got handleChildChange with null currentChildren");
         return;
       }
-      // debug
-      StringBuilder sb = new StringBuilder();
-      for (String child : currentChildren) {
-        sb.append(child).append(",");
-      }
-      LOG.info("list of children in the barrier = " + parentPath + ":" + sb.toString());
-      sb = new StringBuilder();
-      for (String child : names) {
-        sb.append(child).append(",");
-      }
-      LOG.info("list of children to compare against = " + parentPath + ":" + sb.toString());
 
+      LOG.info("list of children in the barrier = " + parentPath + ":" + Arrays.toString(currentChildren.toArray()));
+      LOG.info("list of children to compare against = " + parentPath + ":" + Arrays.toString(names.toArray()));
 
       // check if all the names are in
-      for (String n : names) {
-        if (!currentChildren.contains(n)) {
-          LOG.info("node " + n + " is still not in the list ");
+      for (String name : names) {
```

Comment:
We may consider using Apache CollectionUtil.containsAll(currentChildren, names) for allIn